docs/src/guides/scalar_index.md

```diff
@@ -0,0 +1,108 @@
+# Building Scalar Index
+
+Similar to many SQL databases, LanceDB supports several types of Scalar indices to accelerate search
+over scalar columns.
+
+- `BTREE`: The most common type is BTREE. This index is inspired by the btree data structure
+  although only the first few layers of the btree are cached in memory.
+  It will perform well on columns with a large number of unique values and few rows per value.
+- `BITMAP`: this index stores a bitmap for each unique value in the column.
+  This index is useful for columns with a finite number of unique values and many rows per value.
+  For example, columns that represent "categories", "labels", or "tags"
+- `LABEL_LIST`: a special index that is used to index list columns whose values have a finite set of possibilities.
+  For example, a column that contains lists of tags (e.g. `["tag1", "tag2", "tag3"]`) can be indexed with a `LABEL_LIST` index.
+
+| Data Type                                     | Filter                           | Index Type   |
+| --------------------------------------------- | -------------------------------- | ------------ |
+| Numeric, String, Temporal                      | `<`, `=`, `>`, `in`, `between`, `is null`                    | `BTREE`      |
+| Boolean, numbers or strings with finite range of values         | `<`, `=`, `>`, `in`, `between`, `is null`   | `BITMAP`     |
```

Comment:
I think you mean "few unique values", right? A sequence of all `u32` values has a finite range, but would be a bad bitmap index, right?
```suggestion
| Boolean, numbers or strings with few unique values        | `<`, `=`, `>`, `in`, `between`, `is null`   | `BITMAP`     |
```

Reply:
https://github.com/lancedb/lancedb/pull/1495#discussion_r1710010068 
@westonpace, let's get an agreement on the wording here?

Reply:
I think in the lancedb API PR you said `where the number of unique values is small (i.e., less than a few thousands)`.  Let's use that.  So maybe:

```
Boolean, numbers or strings with fewer than 1,000 unique values
```

That's probably good for "rule of thumb".  I worry that "small" makes people think "dozens".  In some cases a bitmap index will even make sense if there are 10's of thousands of unique values but I don't know how to explain that better without confusing the user.

Reply:
Done